rpcs3/Emu/Cell/lv2/sys_mmapper.cpp

```diff
@@ -384,6 +389,29 @@ error_code sys_mmapper_unmap_shared_memory(u32 addr, vm::ptr<u32> mem_id)
 	// Acknowledge
 	mem->counter--;
 
+	u64 io_addr = (u64)RSXIOMem.getMappedAddress(addr);
+	if (io_addr != -1u)
+	{
+		if (!rsx::get_current_renderer()->isHLE)
+		{
+			// Unmap-key = PackOneBits(unmap_size) << io_addr;
+			// Unmap-flag = (1 << 32) << (io_addr >> 6);
+			u64	unmap_key = u64((1ull << ((u64)mem->size >> 20)) - 1) << ((io_addr >> 20) & 0x3f);
+			u64 unmap_flag = 0x100000000 << (io_addr >> 26);
+			sys_event_port_send(fxm::get<SysRsxConfig>()->rsx_event_port, 0, unmap_flag, unmap_key);
+			thread_ctrl::wait_for(5000); // give _gcm_intr_thread time to handle the event
```

Comment:
Why the sleep? Is it actually needed?

Reply:
is there a way to know when the gcm thread finishes its job and goes back to waiting for events state?  The sleep is a temporery solution until I find a way to be sure that the gcm thread ended its job before the syscall ended.

Reply:
Nope, theres no way to know. this sleep is unnecessary then